Makefile

```
@@ -73,7 +73,7 @@ clean_all: clean clean_deps clean_vendor
 
 .PHONY: format_api_docs_links
 format_api_docs_links:
-	echo $(DOCS_FILES) | xargs sed -i -E -e 's|https?://(www\.)?crystal-lang.org/api/([A-Z])|https://crystal-lang.org/api/latest/\2|g'
+	echo $(DOCS_FILES) | xargs perl -p -i -e 's@\bhttps?://crystal-lang\.org/api/(?!toplevel|[A-Z])[^ )/]+/([^ )]+\.html\b)@https://crystal-lang.org/api/\1@g'
```

I think the regex is working from a wrong angle. It was already like this before, but when we're already changing it, let's make it better.
It shouldn't make any assumptions about the internal paths that don't represent a version prefix. Those could easily change. If additional paths get into the docs generator, they might not be covered by this regex and would be incorrectly recognized as a version prefix.
Instead of excluding paths, the regex should include paths that match the structure of the versions setup.
Something like this:
```suggestion
	echo $(DOCS_FILES) | xargs perl -p -i -e 's@\bhttps?://crystal-lang\.org/api/(master|nightly|\d+(?:\.\d+)+/[^ )/]*([^ )]+\.html\b)@https://crystal-lang.org/api/\1@g'
```

Any reason you removed the `www` subdomain?

---

Well I think this is the correct angle. The doc generator hasn't gained any other page names in 5+ years, and when it does, we'll be fully aware of it. Meanwhile the regex as is can catch a lot of things that are possibly even mistakes. Or branch names that weren't foreseen, like `main`. I mean, you already wrote `master|nightly`, only one of which exists -- that's somewhat strange. Anyway just not to argue here, I'll just change it.

Seems nobody in the history of the world has ever written `www.crystal-lang.org/api` so that's why I thought it wasn't relevant. Maybe it is better to keep it here, but I didn't want to keep it in the Python script. Well maybe I'll put it back.

---

You're right that the doc generator's structure doesn't change a lot. Maybe I'd like to have that too much 😁
But there's still a differentiation of deployment structure vs. content paths.

`master|nightly` was intentional because they're sometimes confused and we might end up separating master and nightly builds at some point. We're not going to have either as a content path, so it's totally fine.

Outright incorrect paths are not really an issue, they'd be caught by the link checker.

---

I think if we'll be changing to `nightly` urls then we'll notice this along with all the other places that need to be changed. So I didn't put that in. Otherwise, done.